lib/kino/process.ex

```diff
@@ -581,21 +586,98 @@ defmodule Kino.Process do
     Map.put_new(rels, lookup, edge)
   end
 
-  defp traverse_links({rels, _idx, pid_keys}) do
+  defp traverse_links({rels, _idx, resource_keys}) do
     rels_with_links =
-      Enum.reduce(pid_keys, rels, fn {pid, _idx}, rels_with_links ->
-        {:links, links} = process_info(pid, :links)
+      Enum.reduce(resource_keys, rels, fn
+        {pid, _idx}, rels_with_links when is_pid(pid) ->
+          {:links, links} = process_info(pid, :links)
+
+          Enum.reduce(links, rels_with_links, fn link_pid, acc ->
+            add_new_links_to_acc(resource_keys, pid, link_pid, acc)
+          end)
 
-        Enum.reduce(links, rels_with_links, fn link_pid, acc ->
-          add_new_links_to_acc(pid_keys, pid, link_pid, acc)
-        end)
+        _, rels_with_links ->
+          rels_with_links
       end)
 
     rels_with_links
   end
 
-  defp add_new_links_to_acc(pid_keys, pid, link_pid, acc) do
-    case pid_keys do
+  defp maybe_traverse_ets_tables(supervision_tree_data, opts) do
+    if Keyword.get(opts, :render_ets_tables, false) do
+      do_traverse_ets_tables(supervision_tree_data)
+    else
+      supervision_tree_data
+    end
+  end
+
+  defp do_traverse_ets_tables(supervision_tree_data) do
+    active_ets_tables =
+      :ets.all()
+      |> Enum.map(fn table ->
+        table
+        |> :ets.info()
+        |> Map.new()
```

Comment:
It is most likely cheaper to lookup the information that you care about:

```suggestion
        %{owner: :ets.info(table, :owner), heir: :ets.info(table, :heir)}
```

You also need to consider the table can be deleted between `:ets.all` and getting its info. So you need to wrap it in a try catch:

```suggestion
      try do
        %{owner: :ets.info(table, :owner), heir: :ets.info(table, :heir)}
      rescue
        _ -> %{owner: nil, heir: nil}
      end
```